code/modules/library/random_books.dm

```diff
@@ -0,0 +1,68 @@
+/obj/item/weapon/book/manual/random/New()
+	var/static/banned_books = list(/obj/item/weapon/book/manual,/obj/item/weapon/book/manual/random,/obj/item/weapon/book/manual/nuclear,/obj/item/weapon/book/manual/wiki)
+	var/newtype = pick(typesof(/obj/item/weapon/book/manual) - banned_books)
+	new newtype(loc)
+	qdel(src)
+
+/obj/structure/bookcase/random
+	var/category = null
+	var/book_count = 2
+	anchored = 1
+	state = 2
+/obj/structure/bookcase/random/New()
+	..()
+	if(ticker && ticker.current_state >= GAME_STATE_PLAYING)
+		initialize()
+
+/obj/structure/bookcase/random/initialize()
+	if(!book_count || !isnum(book_count))
+		update_icon()
+		return
+	if(!establish_db_connection())
+		if(prob(5))
+			var/obj/item/weapon/paper/P = new(get_turf(loc))
+			P.info = "There once was a book from Nantucket<br>But the database failed us, so f*$! it.<br>I tried to be good to you<br>Now this is an I.O.U<br>If you're feeling entitled, well, stuff it!<br><br><font color='gray'>~</font>"
+		update_icon()
+		return
+	if(category && prob(25)) category = null
+	book_count += pick(-1,-1,0,1,1)
+	if(book_count <= 0)
+		update_icon()
+		return
+
+	var/c = category? " AND category='[sanitizeSQL(category)]'" :""
+	var/DBQuery/query = dbcon.NewQuery("SELECT * FROM [format_table_name("library")] WHERE isnull(deleted)[c] GROUP BY title ORDER BY rand() LIMIT [book_count];") // isdeleted copyright (c) not me

```

Comment:
just noticed this. Somebody could varedit bookcount as well. Because it's not enclosed in quotes (something you can't do with the value of the limit clause) passing it thru sanitize_SQL won't cover it. just check book_count with isnum() to ensure it's a number.